src/iexamine.cpp

```diff
@@ -4158,6 +4158,76 @@ static int get_charcoal_charges( units::volume food )
     return  std::max( charcoal, sm_rack::MIN_CHARCOAL );
 }
 
+static bool is_non_rotten_crafting_component( const item &it )
+{
+    return is_crafting_component( it ) && !it.rotten();
+}
+
+void mill_activate( player &p, const tripoint &examp )
+{
+    const furn_id cur_mill_type = g->m.furn( examp );
+    furn_id next_mill_type = f_null;
+    if( cur_mill_type == f_wind_mill ) {
+        next_mill_type = f_wind_mill_active;
+    } else if( cur_mill_type == f_water_mill ) {
+        next_mill_type = f_water_mill_active;
+    } else {
+        debugmsg( "Examined furniture has action mill_activate, but is of type %s",
+                  g->m.furn( examp ).id().c_str() );
+        return;
+    }
+    bool food_present = false;
+    auto items = g->m.i_at( examp );
+    units::volume food_volume = 0_ml;
+
+    for( size_t i = 0; i < items.size(); i++ ) {
+        auto &it = items[i];
+        if( it.typeId() == "flour" ) {
+            add_msg( _( "This mill already contains flour." ) );
+            add_msg( _( "Remove it before starting the mill again." ) );
+            return;
+        }
+        if( it.has_flag( "MILLABLE" ) ) {
+            food_present = true;
+            food_volume += it.volume();
+            continue;
+        }
+        if( !it.has_flag( "MILLABLE" ) ) {
+            add_msg( m_bad, _( "This rack contains %s, which can't be milled!" ), it.tname( 1,
+                     false ) );
+            add_msg( _( "You remove the %s from the mill." ), it.tname().c_str() );
+            g->m.add_item_or_charges( p.pos(), it );
+            g->m.i_rem( examp, i );
+            return;
```

Comment:
Add a move cost for this. Also consider making it a prompt instead of auto.

Reply:
Logic copied from charcoal smoker 
`if( it.typeId() != "charcoal" && !it.has_flag( "SMOKABLE" ) ) {
      add_msg( m_bad, _( "This rack contains %s, which can't be smoked!" ), it.tname( 1, false ) );
      add_msg( _( "You remove %s from the rack." ), it.tname() );
      g->m.add_item_or_charges( p.pos(), it );
      g->m.i_rem( examp, i );
      return;
}`

Would that need a move cost and a prompt too?

Reply:
Move cost - yes, surely. Prompt (in both cases) - on your decision.